portal/app/api/v1/endpoint/farm_device.py

```diff
@@ -20,4 +28,14 @@ def get_queryset(self, **kwargs):
         return queryset
 
 
+class FarmDataViewSet(viewsets.ModelViewSet):
+    serializer_class = FarmDataSerializer
+    authentication_classes = (CsrfExemptAuth.CsrfExemptSessionAuthentication, TokenAuthentication)
+
+    def get_queryset(self, **kwargs):
+        device_uid = self.request.query_params.get('device_uid', None)
+        start_date = self.request.query_params.get('start_date', None)
+        end_date = self.request.query_params.get('end_date', None)
```

Comment:
If all these params are `None`, what happens to the query below? Are you sure you want to run the query anyway?